### DISCUSSION TITLE
[3.x] German umlauts in asset location

### DISCUSSION THREAD

**Discussion by martinherweg** [craftcms/cms#12207] [Q&A] [answered]
Hey everyone,

I have a site with various multisites where I use the site group in asset fields to save assets for the different sites, the asset volume is AWS S3.

Some of the group names contain german umlauts and it seems like sometimes they are converted and sometimes not.
This results in two different folders on AWS and also broken assets.

Any hint how I could streamline the naming so we only have one folder without the need to touch all asset fields?

Thanks and regards

**Comment by brandonkelly** [comment, marked as answer]
The reason for the inconsistency is that the [voku/portable-ascii](https://github.com/voku/portable-ascii) library (which Craft uses to convert strings to ASCII) has a few [German-specific character mappings](https://github.com/voku/portable-ascii/blob/2.0.1/src/voku/helper/data/ascii_by_languages.php#L2036-L2046). So the exact ASCII conversion result will depend on who is signed in (and their preferred language).

I can see how that would be unexpected, though. So we just changed the behavior a bit, to always have the folder/file name ASCII conversion use the primary site’s language, regardless of who is signed in (84460f937a980f70423bf808c147a610f8fc33f3). That change has been released in Craft 3.7.59 and 4.3.1.

**Reply by martinherweg** [reply]
Thanks Brandon, sounds like this could be the reason yes! We will update next week and I if we still have the problem I'll come back to here.
